django/apps/registry.py

```diff
@@ -344,6 +346,76 @@ def clear_cache(self):
             for model in self.get_models(include_auto_created=True):
                 model._meta._expire_cache()
 
+    @staticmethod
+    def _model_key(model_or_name):
+        """
+        Returns an (app_label, model_name) tuple from a model or string.
+        """
+        if isinstance(model_or_name, six.string_types):
+            app_label, model_name = model_or_name.split(".")
+        else:
+            # It's actually a model class.
+            app_label = model_or_name._meta.app_label
+            model_name = model_or_name._meta.object_name
+        return app_label, model_name
+
+    def lazy_model_operation(self, function, *models_or_names, **kwargs):
```

Comment:
Can you say more about what type of introspection migration-state needs to do, and how this helps that? It's still going to be a callback function either way, which is hard to introspect usefully.

Unless there's a strong concrete reason for it, I would rather simplify the API and remove `**kwargs`, even if it means some callers need to curry or define an inline callback.